media-api/test/lib/ElasticSearchTest.scala

```diff
@@ -46,21 +60,20 @@ class ElasticSearchTest extends FunSpec with BeforeAndAfterAll with Matchers wit
       val searchParams = SearchParams(tier = Internal, uploadedBy = Some(testUser))
       val searchResult = ES.search(searchParams)
       whenReady(searchResult) { result =>
-        result.total shouldBe 8
+        result.total shouldBe images.size
       }
     }
 
     it("ES should return all pictures for readonly tier search") {
       val searchParams = SearchParams(tier = ReadOnly, uploadedBy = Some(testUser))
       val searchResult = ES.search(searchParams)
       whenReady(searchResult) { result =>
-        result.total shouldBe 8
+        result.total shouldBe images.size
       }
     }
   }
 
-  override def afterAll {
-    Await.ready(cleanTestUserImages(), 2.seconds)
-    DateTimeUtils.setCurrentMillisSystem()
-  }
+  override def afterAll():Unit = DateTimeUtils.setCurrentMillisSystem()
+
+  override def afterEach(): Unit = Await.ready(deleteImages(), 5.seconds)
```

Comment:
*sigh* tests were failing stating the number of images returned was greater than `images.size`. This is impossible, so delete everything after each test to be safe.

Reply:
@akash1810 is still looks like more is returned that necessary
https://teamcity.gutools.co.uk/viewLog.html?buildId=159713&buildTypeId=EditorialTools_Grid_MediaApi

Maybe something isn't deleted properly?

There is a technique where you make a different name for each test run and only search for that, not sure how applicable that is.

Reply:
Just ran the tests again and ✅ 

![img](https://media.giphy.com/media/l0MYQFMa4aultG1ri/giphy.gif)